sequencer/src/state.rs

```diff
@@ -51,8 +166,111 @@ impl hotshot_types::traits::state::TestableState for State {
     }
 }
 
-impl Committable for State {
-    fn commit(&self) -> Commitment<Self> {
-        unimplemented!("Not used in sequencing consensus")
+pub type BlockMerkleTree = LightWeightSHA3MerkleTree<Commitment<Header>>;
+pub type BlockMerkleCommitment = <BlockMerkleTree as MerkleTreeScheme>::Commitment;
+
+// New Type for `U256` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Add)]
+pub struct FeeAmount(U256);
+// New Type for `Address` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(
+    Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord,
+)]
+pub struct FeeAccount(Address);
+
+impl Valid for FeeAmount {
+    fn check(&self) -> Result<(), SerializationError> {
+        Ok(())
+    }
+}
+
+impl Valid for FeeAccount {
+    fn check(&self) -> Result<(), SerializationError> {
+        Ok(())
+    }
+}
+
+impl CanonicalSerialize for FeeAmount {
+    fn serialize_with_mode<W: std::io::prelude::Write>(
+        &self,
+        mut writer: W,
+        _compress: Compress,
+    ) -> Result<(), SerializationError> {
+        let mut bytes = [0u8; core::mem::size_of::<U256>()];
+        self.0.to_little_endian(&mut bytes);
+        Ok(writer.write_all(&bytes)?)
+    }
+
+    fn serialized_size(&self, _compress: Compress) -> usize {
+        core::mem::size_of::<U256>()
+    }
+}
+impl CanonicalDeserialize for FeeAmount {
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        _compress: Compress,
+        _validate: Validate,
+    ) -> Result<Self, SerializationError> {
+        let mut bytes = [0u8; core::mem::size_of::<U256>()];
+        reader.read_exact(&mut bytes)?;
+        let value = U256::from_little_endian(&bytes);
+        Ok(Self(value))
+    }
+}
+impl CanonicalSerialize for FeeAccount {
+    fn serialize_with_mode<W: std::io::prelude::Write>(
+        &self,
+        mut writer: W,
+        _compress: Compress,
+    ) -> Result<(), SerializationError> {
+        Ok(writer.write_all(self.0.as_bytes())?)
+    }
+
+    fn serialized_size(&self, _compress: Compress) -> usize {
+        core::mem::size_of::<Address>()
     }
 }
+impl CanonicalDeserialize for FeeAccount {
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        _compress: Compress,
+        _validate: Validate,
+    ) -> Result<Self, SerializationError> {
+        let mut bytes = [0u8; core::mem::size_of::<Address>()];
+        reader.read_exact(&mut bytes)?;
+        let value = Address::from_slice(&bytes);
+        Ok(Self(value))
+    }
+}
+impl std::convert::From<u64> for FeeAccount {
```

Comment:
@tbro is this impl used? If not, what is it for?

Reply:
It is only there to comply with the bounds of UniversalMerkleTree.  I believe that bound has since been removed though, so we can probably remove it.

Reply:
@sveitser I'm currently working on #1046 , which will render some of these questions irrelevant. Now that this PR is merged I suggest you save your time for future reviews.